machinist/mserver/controller.go

```diff
@@ -163,3 +166,18 @@ func (en *Controller) ServeAllRecords() {
 		_ = <-time.After(5 * time.Second)
 	}
 }
+
+// WriteState writes state to the specified state file every 2 seconds. Will not exit or error out unless no statefile is
+// provided.
+func (en *Controller) WriteState() {
+	if en.stateFile == "" {
+		en.Log.Warnf("No path to state provided, state is fully in memory")
```

Comment:
important: don't you need a return here? Otherwise the loop below will keep running, and fail forever?

Reply:
good catch, it passed the test because you cant write to ""